apps/addons/models.py

```diff
@@ -541,6 +541,8 @@ def get_url_path(self, more=False, add_prefix=True):
         # If more=True you get the link to the ajax'd middle chunk of the
         # detail page.
         view = 'addons.detail_more' if more else 'addons.detail'
+        if not self.is_listed:
+            view = 'devhub.addons.edit'
```

Comment:
We can discuss this some more if you want, i've done that following an irc discussion with @clouserw. Wil, can you please confirm?